benchmarking/commons/hpo_main_simulator.py

```diff
@@ -365,8 +352,70 @@ def main(
             print_update_interval=600,
             tuner_name=tuner_name,
             metadata=metadata,
-            save_tuner=args.save_tuner,
+            save_tuner=configuration.save_tuner,
         )
         tuner.run()
         if post_processing is not None:
             post_processing(tuner)
+
+
+def main(
+    methods: MethodDefinitions,
+    benchmark_definitions: SurrogateBenchmarkDefinitions,
+    extra_args: Optional[ExtraArgsType] = None,
+    map_extra_args: Optional[MapMethodArgsType] = None,
+    post_processing: Optional[PostProcessingType] = None,
+    use_transfer_learning: bool = False,
+):
+    """
+    Runs sequence of experiments with simulator backend sequentially. The loop
+    runs over methods selected from ``methods``, repetitions and benchmarks
+    selected from ``benchmark_definitions``, with the range being controlled by
+    command line arguments.
+
+    ``map_method_args`` can be used to modify ``method_kwargs`` for constructing
+    :class:`~benchmarking.commons.baselines.MethodArguments`, depending on
+    ``configuration`` and the method. This allows for extra flexibility to specify specific arguments for chosen methods
+    Its signature is :code:`method_kwargs = map_method_args(configuration, method, method_kwargs)`,
+    where ``method`` is the name of the baseline.
+
+    :param methods: Dictionary with method constructors
+    :param benchmark_definitions: Definitions of benchmarks
+    :param extra_args: Extra arguments for command line parser. Optional
+    :param map_extra_args: Maps ``args`` returned by :func:`parse_args` to dictionary
+        for extra argument values. Needed if ``extra_args`` given
+    :param post_processing: Called after tuning has finished, passing the tuner
+        as argument. Can be used for postprocessing, such as output or storage
+        of extra information
+    :param use_transfer_learning: If True, we use transfer tuning. Defaults to
+        False
+    """
+    simulated_backend_extra_parameters = SIMULATED_BACKEND_EXTRA_PARAMETERS.copy()
+    if is_dict_of_dict(benchmark_definitions):
+        simulated_backend_extra_parameters.append(BENCHMARK_KEY_EXTRA_PARAMETER)
+    configuration = config_from_argparse(extra_args, simulated_backend_extra_parameters)
+
+    configuration.support_checkpointing = bool(configuration.support_checkpointing)
```

Comment:
Don't need that. Please grep through the code for "bool(" and check whether further such leftovers are there.